tools/SourceKit/tools/sourcekitd/bin/XPC/Service/XPCService.cpp

```diff
@@ -207,6 +207,29 @@ std::string sourcekitd::getRuntimeLibPath() {
   return Path.str();
 }
 
+std::string sourcekitd::getDiagnosticDocumentationPath() {
+  llvm::SmallString<128> Path = StringRef(llvm::sys::fs::getMainExecutable(
+      "sourcekit",
+      reinterpret_cast<void *>(&anchorForGetMainExecutableInXPCService)));
+#ifdef SOURCEKIT_UNVERSIONED_FRAMEWORK_BUNDLE
```

Comment:
We should factor the nesting level stuff out so it can be shared with the runtime path.  I suggest a method for getting the prefix path (i.e. "usr") that both can append to.

Reply:
Good idea, I pulled that out into a `getToolchainUsrPath` helper, similar to the in-process version.